### PR TITLE
Support case sensitive regex filter 

### PR DESCRIPTION
**Pull request by tuxinaut**
Implements #1485

We're aware of the Contributor agreement and our legal team is working on it (eta this week)

### CODE REVIEW

**Comment by kapilt** [comment]
Thanks for the pull request, we've had a few cases where legal on cla takes months, or never, so preference would be to have the pr after thats dealt with.

**Comment by kapilt** [comment]
closing for now, please re-open/comment when cla has been addressed, note on a corporate cla signature we still need individual cla signatures (as contributing members often change over time).

**Comment by tuxinaut** [comment]
Our legal team already approved the Contributor agreement so please reopen the PR.

File: c7n/filters/core.py [hunk 1]
```diff
@@ -90,6 +98,7 @@ def intersect(x, y):
     'less-than': operator.lt,
     'glob': glob_match,
     'regex': regex_match,
+    'regex-cs': regex_case_sensitive_match,
```
Comment (kapilt) [inline c7n/filters/core.py]: thoughts on renaming to `regex-case`?... cs feels a bit ambigious without context.

File: c7n/filters/core.py [hunk 2]
```diff
@@ -422,7 +431,7 @@ def validate(self):
             if not self.data['op'] in OPERATORS:
                 raise PolicyValidationError(
                     "Invalid operator in value filter %s" % self.data)
-            if self.data['op'] == 'regex':
+            if self.data['op'] in {'regex', 'regex-cs'}:
```
Comment (kapilt) [inline c7n/filters/core.py]: i did a quick benchmark and was surprised by the efficiency of the set here vs tuple even for low cardinalities.

**Comment by tuxinaut** [comment]
@kapilt completed the form 1 minuted ago. Thx for your patient :smile:

**Comment by kalrish** [comment]
I renamed the filter to `regex-case`, rebasing the PR so that the history stays clean.

**Comment by kalrish** [comment]
I also signed the individual CLA :smile:

**Pull request merged**